client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx

```diff
@@ -7,10 +7,13 @@ export default function DomainAndPlanPackageNavigation( props ) {
 	const translate = useTranslate();
 
 	const goBack = () => {
+		if ( window.history.length > 1 ) {
+			window.history.go( -1 );
+			return;
+		}
+
 		if ( props.goBackLink ) {
 			window.location.assign( props.goBackLink );
-		} else {
-			window.history.go( -1 );
 		}
```

Comment:
This change _seems_ to work, but `history.length` isn't reduced after `history.go( -1 )`, which means that if a user reaches the `Select a plan` step, we'll always run this code instead of `location.assign()`.

Maybe it's safer to revert this specific change..?

Reply:
I'm comfortable with that. The motivation for this change was the 
> It should link to the previous page if possible. If not, go to `:site/wp-admin` if the site is configured to the classic interface. Otherwise, go to `/home/:site`

portion of the related issue's acceptance criteria, but it did feel a little odd to me that we'd virtually never, or just never never, actually make it to the `location.assign` flow.

If we revert this change, I'd probably also opt to revert the change to the button text, and make it say **Home** by default, and conditionally say **Back** if that path is utilized (which is what it does currently on production)

Effectively, this PR would really just make sure the **Home** path is conditional based on the user's interface preference, but leave all other behavior as it is.

That all works for me. Let me know if you have any objections/concerns @arthur791004 🙂